torch/_dynamo/eval_frame.py

```diff
@@ -631,22 +634,28 @@ def output(self, target, args, kwargs):
             new_result_flat = [lookup[i] for i in matched_output_elements_positions]
             return super().output(target, (new_result_flat,), {})
 
-        def run_node(self, n):
-            self.current_node = n
-            return super().run_node(n)
-
     if aten_graph:
         # Running graph with interpreter is needed for propagating the stack_trace
         def graph_with_interpreter(*args):
             with torch.fx.traceback.preserve_node_meta():
                 return torch.fx.Interpreter(graph).run(*args)
 
-        graph = make_fx(
-            graph_with_interpreter,
-            decomposition_table=decomposition_table,
-            tracing_mode=tracing_mode,
-            _allow_non_fake_inputs=True,
-        )(*graph_captured_input)
+        if tracing_mode == "real":
+            graph = make_fx(
+                graph_with_interpreter,
+                decomposition_table=decomposition_table,
+            )(*graph_captured_input)
+        elif tracing_mode == "symbolic":
+            # For dynamic shape, we need to make_fx through the graph with fake tensors under FakeTensorMode
+            # The fake tensors may contain the fine grain dynamic shape passed down from dynamo
+            fake_mode = fake_mode_from_tensors(compile_time_inputs)
+            with fake_mode:
+                graph = make_fx(
+                    graph_with_interpreter,
+                    decomposition_table=decomposition_table,
+                )(*compile_time_inputs)
```

Comment:
Can fake mode be a requirement for symbolic in make_fx? Can we push this down?

Reply:
Notice that, this is calling make_fx with "real" tracing mode using FakeTensor, not "symbolic" tracing mode with FakeTensor. 

I tried pushing FakeMode into make_fx, but I couldn't get it passing the UTs. 
Here's what I have tried: in make_fx, if the tracing model is "real" and if any of the input tensor is a FakeTensor, I set the fakemode.

Reply:
This is what we do today in aot_autograd. Inside aotautograd, we always ignore user specified tracing mode, and always use the tracing_mode="real". According to @Chillee, it's also expected to use tracing_mode="real" internally. 

See
https://github.com/pytorch/pytorch/blob/master/torch/_functorch/aot_autograd.py#L1053
and
https://github.com/pytorch/pytorch/blob/master/torch/_functorch/aot_autograd.py#L1643

I tried changing the behavior of make_fx, but make_fx is used in so many places, and it's breaks bunch of tests, and it appears that there are subtly of how FakeTensor behave differently under FakeTensorMode vs. non-FakeTensorMode. 

So to me, the easiest path forward is to replicate what AOTAutograd is doing in dynamo.export.